cli/flox/src/commands/show.rs

```diff
@@ -126,7 +137,70 @@ fn construct_show_params(
     Ok(search_params)
 }
 
-fn render_show(search_results: &[SearchResult]) -> Result<()> {
+fn render_show_catalog(
+    search_results: &[SearchResult],
+    expected_systems: &HashSet<System>,
+) -> Result<()> {
+    if search_results.is_empty() {
+        // This should never happen since we've already checked that the
+        // set of results is non-empty.
+        bail!("no packages found");
+    }
+    let pkg_name = search_results[0].rel_path.join(".");
+    let description = search_results[0]
+        .description
+        .as_ref()
+        .map(|d| d.replace('\n', " "))
+        .unwrap_or(DEFAULT_DESCRIPTION.into());
+    println!("{pkg_name} - {description}");
+
+    // Organize the versions to be queried and printed
+    let version_to_systems = {
+        let mut map = BTreeMap::new();
+        for pkg in search_results.iter() {
+            if let Some(ref version) = pkg.version {
+                map.entry(version.clone())
+                    .or_insert(HashSet::new())
+                    .insert(pkg.system.clone());
+                // map.insert(version.clone(), pkg.system.clone());
+            }
+        }
+        map
+    };
+    let mut seen_versions = HashSet::new();
+    for pkg in search_results {
+        if let Some(ref version) = pkg.version {
+            if seen_versions.contains(&version) {
+                // We print everything in one go for each version, so if we've seen it once
+                // we don't need to do anything else.
+                continue;
+            }
+            let Some(systems) = version_to_systems.get(version) else {
+                continue;
+            };
+            let available_systems = {
+                let mut intersection = expected_systems
+                    .intersection(systems)
+                    .cloned()
+                    .collect::<Vec<_>>();
+                intersection.sort();
+                intersection
+            };
+            if available_systems.len() != expected_systems.len() {
+                println!(
+                    "    {pkg_name}@{version} ({} only)",
+                    available_systems.join(", ")
+                );
+            } else {
+                println!("    {pkg_name}@{version}");
+            }
+            seen_versions.insert(version);
+        }
```

Comment:
nonblocking: looks like version is required in the response from the server. Looks like we should add a TODO about making version not an `Optional` here? I thought at first we might be dropping packages

Reply:
It's optional from pkgdb but not from the server but they both get turned into `SearchResult` for the moment. I drop the non-existent packages from the server that have no version field. Once we drop pkgdb searching we can reformat a few things, namely not converting `show` results to `SearchResult` since that no longer has the same format as a search result anymore.